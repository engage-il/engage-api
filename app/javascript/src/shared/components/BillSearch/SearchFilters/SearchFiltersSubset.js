// @flow
import * as React from 'react'
import styled from 'react-emotion'
import { mixins, colors } from '@/styles'
import type { SearchParamsSubset } from '@/types'

type SubsetEntry = {
  label: string,
  value: SearchParamsSubset,
}

type Props = {
  entry: SubsetEntry,
  value: ?SearchParamsSubset,
  onChange: (event: SyntheticInputEvent) => void
}

export function SearchFiltersSubset ({
  entry,
  value,
  onChange
}: Props) {
  return (
    <Option>
      <input
        type='checkbox'
        name='subset'
        value={entry.value}
        onChange={onChange}
        checked={entry.value === value}
      />
      <span
        children={entry.label}
      />
    </Option>
  )
}

const Option = styled.label`
  ${mixins.flexRow};
  ${mixins.fonts.regularSlab};
  ${mixins.fontSizes.h5};

  align-items: center;
  color: ${colors.gray2};

  > input {
    margin-right: 10px;
  }
`
