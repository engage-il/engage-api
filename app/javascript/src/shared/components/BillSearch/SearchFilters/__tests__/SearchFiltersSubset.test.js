/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { defaultsDeep } from 'lodash'
import { SearchFiltersSubset } from '../SearchFiltersSubset'

// subject
let subject

const defaults = {
  entry: {
    label: 'test-label',
    value: 'test-value'
  },
  value: 'test-value',
  onChange: jest.fn()
}

function loadSubject (props = {}) {
  subject = shallow(<SearchFiltersSubset {...defaultsDeep(props, defaults)} />)
}

// specs
afterEach(() => {
  subject = null
})

describe('#render', () => {
  it('shows the subset filter', () => {
    loadSubject()
    expect(subject).toMatchSnapshot()
  })
})
