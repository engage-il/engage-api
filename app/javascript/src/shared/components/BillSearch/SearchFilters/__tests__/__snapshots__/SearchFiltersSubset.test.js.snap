// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render shows the subset filter 1`] = `
<Styled(label)>
  <input
    checked={true}
    name="subset"
    onChange={[Function]}
    type="checkbox"
    value="test-value"
  />
  <span>
    test-label
  </span>
</Styled(label)>
`;
